### PR TITLE
mro h7 secured firmware

### PR DESCRIPTION
**Pull request by jrn90**
This PR is for automating the security and testing of the mro-h7 flight controller. More info found [here](https://github.com/render-se/SRA/issues/69).

## Acceptance Criteria
- [x] `make cppcheck_secure` make target created & runs
- [x] `make scan-build_secure` make target created & runs
- [x] `make clang-tidy_secure` make target created & runs
- [x] `make coverage_secure` make target created & runs

### CODE REVIEW

File: .github/workflows/rse-checks.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
+name: RSE Checks and Targets
```
Comment (bpmooch) [inline .github/workflows/rse-checks.yml:1]: great job on this workflow

File: boards/mro/sitl/default.px4board [hunk 2]
```diff
@@ -0,0 +1,75 @@
+CONFIG_PLATFORM_POSIX=y
```
Comment (bpmooch) [inline boards/mro/sitl/default.px4board:1]: Did you copy the sitl target as the basis for this target? What had to be changed?

File: platforms/common/include/px4_platform_common/param.h [hunk 3]
```diff
@@ -73,7 +73,7 @@ inline static param_t param_handle(px4::params p)
 
 // Define a list of parameters. This macro also creates code to update parameters.
 // If you get a compile error like:
-//   error: virtual function ‘virtual void <class>::updateParamsImpl()’
+//   error: virtual function 'virtual void <class>::updateParamsImpl()'
```
Comment (bpmooch) [inline platforms/common/include/px4_platform_common/param.h]: why did this change? we need to avoid making superficial changes to the fork

Comment (bpmooch) [reply platforms/common/include/px4_platform_common/param.h]: there are a few examples of this actually

**Review by bpmooch — CHANGES_REQUESTED** [review]
Looks mostly good, but the text changes need to be fixed

File: platforms/common/include/px4_platform_common/param.h (hunk 3 above)
Comment (jrn90) [reply platforms/common/include/px4_platform_common/param.h]: This is actually a fix for cppcheck errors. The report fails to generate due to encoding. These characters are outside of the "normal" (using that very loosely) ASCII encoding the report is using.

File: .github/workflows/rse-checks.yml (hunk 1 above)
Comment (jrn90) [reply .github/workflows/rse-checks.yml:1]: ty ty :rocket: 

I think there are some improvements that could be done to this but I think it is in a decent enough spot.

File: boards/mro/sitl/default.px4board (hunk 2 above)
Comment (jrn90) [reply boards/mro/sitl/default.px4board:1]: Yup, it was too much to go the other way around. Not much needed to be changed due to the way sitl has to "fake" a lot of the IO.

**Comment by jrn90** [comment]
Fixed cppcheck-htmlreport errors and reverted text changes.

Also rebased from master to new rse branch.

**Review by bpmooch — APPROVED** [review]
lgtm, needs squash

**Pull request merged**